File: app/src/main/java/com/rp/picture/cameraImage/OnPicker.java
package com.rp.picture.cameraImage;

import static android.os.Build.VERSION_CODES.M;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.graphics.Point;
import android.graphics.drawable.ColorDrawable;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.os.Parcel;
import android.os.Parcelable;
import android.view.Display;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.android.material.bottomsheet.BottomSheetDialog;
import com.rp.picture.R;
import com.rp.uihelpher.log.IsLog;

import java.io.File;
import java.io.Serializable;

public class OnPicker extends Fragment {

    private String TAG = OnPicker.class.getSimpleName();
    private OnPickerListener onPickerListener;
    private Activity mActivity;
    private String selectLang;
    TextView tv_header,tv_camera,tv_gallery,tv_cancel;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mActivity = getActivity();
//        onDlg();
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mActivity = getActivity();

        if (getArguments() != null) {
            selectLang = getArguments().getString("selectLang");
            onPickerListener = (OnPickerListener) getArguments().getSerializable("listener");
        }
    }

    // Dashboard
    public OnPicker(Activity mActivity,String selectLang, OnPickerListener listener) {
        this.mActivity = mActivity;
        this.selectLang = selectLang;
        onPickerListener = listener;

        onDlg();
    }

        protected void onDlg() {

            final BottomSheetDialog dialog = new BottomSheetDialog(mActivity);
            LayoutInflater inflater = mActivity.getLayoutInflater();
            View convertView = inflater.inflate(R.layout.dlg_pic_opt, null);

            tv_header = convertView.findViewById(R.id.tv_header);
            tv_camera = convertView.findViewById(R.id.tv_camera);
            tv_gallery = convertView.findViewById(R.id.tv_gallery);
            tv_cancel = convertView.findViewById(R.id.tv_cancel);

            convertView.findViewById(R.id.cv_bg).setBackground(mActivity.getResources().getDrawable(R.drawable.bg_cv_dlg));

            convertView.findViewById(R.id.tv_cancel).setOnClickListener(view -> dialog.dismiss());

            convertView.findViewById(R.id.ll_gallery).setOnClickListener(view -> {
                Intent intent = new Intent(mActivity, ImgSelectAct.class);
                intent.putExtra(ImgSelectAct.FLAG_COMPRESS, true);
                intent.putExtra(ImgSelectAct.FLAG_CAMERA, true);
                intent.putExtra(ImgSelectAct.FLAG_GALLERY, true);
                intent.putExtra(ImgSelectAct.FLAG_CAMERA_YES_NO, false);
//                startActivityForResult(intent, selectPage);
                someActivityResultLauncher.launch(intent);

                dialog.dismiss();
            });
            convertView.findViewById(R.id.ll_camera).setOnClickListener(view -> {
                Intent intent = new Intent(mActivity, ImgSelectAct.class);
                intent.putExtra(ImgSelectAct.FLAG_COMPRESS, true);
                intent.putExtra(ImgSelectAct.FLAG_CAMERA, true);
                intent.putExtra(ImgSelectAct.FLAG_GALLERY, true);
                intent.putExtra(ImgSelectAct.FLAG_CAMERA_YES_NO, true);
                someActivityResultLauncher.launch(intent);
                dialog.dismiss();
            });

            dialog.getWindow().setBackgroundDrawable(
                    new ColorDrawable(Color.TRANSPARENT));
            WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
            Window window = dialog.getWindow();
            lp.copyFrom(window.getAttributes());

            Display display = mActivity.getWindowManager().getDefaultDisplay();
            Point size = new Point();
            display.getSize(size);
            lp.width = size.x;
            lp.height = size.y;
            // This makes the dialog take up the full width
            window.setAttributes(lp);
            dialog.setCancelable(true);
            dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
            dialog.setContentView(convertView);
            dialog.setTitle("");
            dialog.show();
        }

    ActivityResultLauncher<Intent> someActivityResultLauncher = registerForActivityResult(
            new ActivityResultContracts.StartActivityForResult(),
            new ActivityResultCallback<>() {
                @Override
                public void onActivityResult(ActivityResult result) {
                    if (result.getResultCode() == Activity.RESULT_OK) {
                        Intent data = result.getData();
                        try {
                            assert data != null;
                            String filePath = data.getStringExtra(ImgSelectAct.RESULT_FILE_PATH);
                            Uri imageUri = Uri.parse("");
                            if (Build.VERSION.SDK_INT >= M) {
                                imageUri = Uri.fromFile(new File(filePath));
                                onPickerListener.onViewImg(imageUri.toString(), "");
                            } else {
                                Bitmap selectedImage = BitmapFactory.decodeFile(filePath);
                                imageUri = Uri.fromFile(SaveBitmap.save(selectedImage));
                                onPickerListener.onViewImg(imageUri.toString(), "");
                            }

                            new IsLog(TAG, "imageUri======" + imageUri.toString());
                        } catch (NumberFormatException | NullPointerException e) {
                            new IsLog(TAG, "NumberFormatException " + e.getMessage());
                        } catch (Exception e) {
                            new IsLog(TAG, "Exception " + e.getMessage());
                        }
                    }
                }
            });

    public interface OnPickerListener extends Parcelable, Serializable {

        void onViewImg(String imageUri,String opt);

        @Override
        default int describeContents() {
            return 0;
        }

        @Override
        default void writeToParcel(@NonNull Parcel dest, int flags) {

        }

        Creator<OnPickerListener> CREATOR = new Creator<OnPickerListener>() {
            @Override
            public OnPickerListener createFromParcel(Parcel in) {
                return (imageUri, opt) -> {

                };
            }

            @Override
            public OnPickerListener[] newArray(int size) {
                return new OnPickerListener[size];
            }
        };
    }
}

File: app/src/main/java/com/rp/picture/cameraImage/OnPicker.kt
package com.rp.picture.cameraImage

import android.app.Activity
import android.content.Intent
import android.graphics.BitmapFactory
import android.graphics.Color
import android.graphics.Point
import android.graphics.drawable.ColorDrawable
import android.net.Uri
import android.os.Build
import android.os.Build.VERSION_CODES
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.os.Parcel
import android.os.Parcelable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.view.WindowManager
import android.widget.TextView
import androidx.activity.result.ActivityResult
import androidx.activity.result.ActivityResultCallback
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
import androidx.fragment.app.Fragment
import com.google.android.material.bottomsheet.BottomSheetDialog
import com.rp.picture.R
import com.rp.picture.cameraImage.SaveBitmap.save
import com.rp.picture.onPermission.OnGlobalPermission
import com.rp.picture.onPermission.OnGlobalPermission.Companion.onDialogShow
import com.rp.uihelpher.log.IsLog
import java.io.File
import java.io.Serializable

class OnPicker() : Fragment() {

    private val TAG = OnPicker::class.java.simpleName

    private var onPickerListener: OnPickerListener? = null
    private var mActivity: Activity? = null
    private var selectLang: String? = null

    private lateinit var tv_header: TextView
    private lateinit var tv_camera: TextView
    private lateinit var tv_gallery: TextView
    private lateinit var tv_cancel: TextView

    constructor(mActivity: Activity, selectLang: String, listener: OnPickerListener) : this() {

        this.mActivity = mActivity
        this.selectLang = selectLang
        this.onPickerListener = listener

        if (onDialogShow == true) {
            onDlg()
        } else {

            Handler(Looper.getMainLooper()).postDelayed({

                val intent = Intent(mActivity, ImgSelectAct::class.java)
                intent.putExtra(ImgSelectAct.FLAG_COMPRESS, true)
                intent.putExtra(ImgSelectAct.FLAG_CAMERA, true)
                intent.putExtra(ImgSelectAct.FLAG_GALLERY, true)
                intent.putExtra(ImgSelectAct.FLAG_CAMERA_YES_NO, false)

                someActivityResultLauncher.launch(intent)

            }, 200)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mActivity = activity

        arguments?.let {
            selectLang = it.getString("selectLang")
            onPickerListener = it.getSerializable("listener") as? OnPickerListener
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        mActivity = activity
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    private fun onDlg() {

        mActivity?.let { mActivity ->
            val dialog = BottomSheetDialog(mActivity)
            val inflater = mActivity.layoutInflater

            val convertView = inflater.inflate(R.layout.dlg_pic_opt, null)

            tv_header = convertView.findViewById(R.id.tv_header)
            tv_camera = convertView.findViewById(R.id.tv_camera)
            tv_gallery = convertView.findViewById(R.id.tv_gallery)
            tv_cancel = convertView.findViewById(R.id.tv_cancel)

            if (selectLang == "AR") {
                tv_header.text = OnGlobalPermission.TAG_SELECT_SOURCE_ARA
                tv_camera.text = OnGlobalPermission.TAG_CAMERA_ARA
                tv_gallery.text = OnGlobalPermission.TAG_GALLERY_ARA
                tv_cancel.text = OnGlobalPermission.TAG_CANCEL_ARA
            } else {
                tv_header.text = OnGlobalPermission.TAG_SELECT_SOURCE_ENG
                tv_camera.text = OnGlobalPermission.TAG_CAMERA_ENG
                tv_gallery.text = OnGlobalPermission.TAG_GALLERY_ENG
                tv_cancel.text = OnGlobalPermission.TAG_CANCEL_ENG
            }

            convertView.findViewById<View>(R.id.cv_bg)
                .background = resources.getDrawable(R.drawable.bg_cv_dlg)

            convertView.findViewById<View>(R.id.tv_cancel)
                .setOnClickListener { dialog.dismiss() }

            convertView.findViewById<View>(R.id.ll_gallery)
                .setOnClickListener {

                    val intent = Intent(mActivity, ImgSelectAct::class.java)
                    intent.putExtra(ImgSelectAct.FLAG_COMPRESS, true)
                    intent.putExtra(ImgSelectAct.FLAG_CAMERA, true)
                    intent.putExtra(ImgSelectAct.FLAG_GALLERY, true)
                    intent.putExtra(ImgSelectAct.FLAG_CAMERA_YES_NO, false)

                    someActivityResultLauncher.launch(intent)
                    dialog.dismiss()
                }

            convertView.findViewById<View>(R.id.ll_camera)
                .setOnClickListener {

                    val intent = Intent(mActivity, ImgSelectAct::class.java)
                    intent.putExtra(ImgSelectAct.FLAG_COMPRESS, true)
                    intent.putExtra(ImgSelectAct.FLAG_CAMERA, true)
                    intent.putExtra(ImgSelectAct.FLAG_GALLERY, true)
                    intent.putExtra(ImgSelectAct.FLAG_CAMERA_YES_NO, true)

                    someActivityResultLauncher.launch(intent)
                    dialog.dismiss()
                }

            dialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))

            val lp = WindowManager.LayoutParams()
            val window = dialog.window
            lp.copyFrom(window?.attributes)
            val display = mActivity.windowManager.defaultDisplay
            val size = Point()
            display.getSize(size)
            lp.width = size.x
            lp.height = size.y
            window?.attributes = lp
            dialog.setCancelable(true)
            dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
            dialog.setContentView(convertView)
            dialog.show()
        }

    }

    private val someActivityResultLauncher = registerForActivityResult(StartActivityForResult()) { result ->

            if (result.resultCode == Activity.RESULT_OK) {

                val data = result.data

                try {

                    val filePath =
                        data?.getStringExtra(ImgSelectAct.RESULT_FILE_PATH)

                    var imageUri = Uri.parse("")

                    imageUri = Uri.fromFile(File(filePath))
                    onPickerListener?.onViewImg(imageUri.toString(), "")

                    IsLog(TAG, "imageUri======$imageUri")

                } catch (e: Exception) {

                    IsLog(TAG, "Exception ${e.message}")
                }
            }
        }

    fun interface OnPickerListener : Parcelable, Serializable {

        fun onViewImg(imageUri: String, opt: String)

        override fun describeContents(): Int = 0

        override fun writeToParcel(dest: Parcel, flags: Int) {}

        companion object CREATOR : Parcelable.Creator<OnPickerListener> {

            override fun createFromParcel(source: Parcel
            ): OnPickerListener {
                return OnPickerListener { _, _ -> }
            }

            override fun newArray(size: Int): Array<OnPickerListener?> {
                return arrayOfNulls(size)
            }
        }
    }
}
